lib/matplotlib/legend_handler.py

```diff
@@ -649,11 +661,37 @@ class HandlerTuple(HandlerBase):
     pad : float, optional
         If None, fall back to ``legend.borderpad`` as the default.
         In units of fraction of font size. Default is None.
+
+
+    width_ratios : tuple, optional
+        Specifies the respective widths of a text/arrow legend annotation pair.
+        Must be of length ndivide.
+        If None, all sections will have the same width. Default is None.
+
+
+    handlers : tuple, optionnal
+        The list of handlers to call for each text/arrow legend annotation section.
+        Must be of length ndivide.
+        If None, the default handlers will be fetched automatically. Default is None.
     """
-    def __init__(self, ndivide=1, pad=None, **kwargs):
+
+    def __init__(
+            self,
+            ndivide=1,
+            pad=None,
+            width_ratios=None,
+            handlers=None,
+            **kwargs):
 
         self._ndivide = ndivide
         self._pad = pad
+        self._handlers = handlers
```

Comment:
Can you validate that `handlers` is the correct length?  That way we can raise an exception as early as possible.